content/token22-in-the-client.md

```diff
@@ -0,0 +1,470 @@
+
+Here are the arguments we'll be passing into this function:
+
+- `connection` - The connection object to use
+- `tokenProgramId` - The token program to point to
+- `payer` - The keypair paying for the transaction
+- `decimals` - The number of decimals to include for the mint
+- `mintAmount` - The amount of tokens to mint to the payer
+
+And this is what the function will do:
+
+- Create a new mint using **`createMint`**
+- Fetch mint information using **`getMint`**
+- Log mint information using **`printTableData`**
+- Create an associated token account with **`getOrCreateAssociatedTokenAccount`**
+- Log the address of the associated token account
+- Mint tokens to the associated token account with **`mintTo`**
+
+All put together this is what the final `createAndMintToken` function looks like:
+
+```ts
+import { createMint, getMint, getOrCreateAssociatedTokenAccount, mintTo } from "@solana/spl-token";
+import { Connection, Keypair, PublicKey } from "@solana/web3.js";
+import printTableData from "./print-helpers";
+
+export async function createAndMintToken(
+	connection: Connection,
+	tokenProgramId: PublicKey,
+	payer: Keypair,
+	decimals: number,
+	mintAmount: number,
+): Promise<PublicKey> {
+	console.log('\\nCreating a new mint...')
+	const mint = await createMint(
+		connection,
+		payer,
+		payer.publicKey,
+		payer.publicKey,
+		decimals,
+		undefined,
+		{
+			commitment: 'finalized', // confirmOptions argument
+		},
+		tokenProgramId
+	)
+
+	console.log('\\nFetching mint info...')
+
+	const mintInfo = await getMint(
+		connection,
+		mint,
+		'finalized',
+		tokenProgramId
+	)
+
+	printTableData(mintInfo);
+
+	console.log('\\nCreating associated token account...')
+	const tokenAccount = await getOrCreateAssociatedTokenAccount(
+		connection,
+		payer,
+		mint,
+		payer.publicKey,
+		true,
+		'finalized',
+		{commitment: 'finalized'},
+		tokenProgramId
+	)
+
+	console.log(`Associated token account: ${tokenAccount.address.toBase58()}`)
+
+	console.log('\\nMinting to associated token account...')
+	await mintTo(
+		connection,
+		payer,
+		mint,
+		tokenAccount.address,
+		payer,
+		mintAmount,
+		[payer],
+		{commitment: 'finalized'},
+		tokenProgramId
+	)
+
+	return mint
+}
+
+export default createAndMintToken
+```
+
+
+
+### 5. Creating and minting tokens
+
+Let's now take our new function and invoke it twice within our main script in `index.ts`. We'll want a `Token Program` and `Token Extension Program` token to test against. So we'll use our two different program IDs:
+
+```tsx
+import { initializeKeypair } from '@solana-developers/helpers';
+import { Cluster, Connection } from '@solana/web3.js';
+import createAndMintToken from './create-and-mint-token';
+import printTableData from './print-helpers';
+import { TOKEN_2022_PROGRAM_ID, TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import dotenv from 'dotenv';
+dotenv.config();
+
+(async () => {
```

Comment:
`esrun` also means you can skip the `IIFEs` too.

Reply:
We'll refactor!